lib/filterx/expr-condition.c

```diff
@@ -55,6 +56,26 @@ _eval(FilterXExpr *s)
   if (!condition_value)
     return NULL;
 
+  if (trace_flag)
+    {
+      ScratchBuffersMarker mark;
+      GString *buf = scratch_buffers_alloc_and_mark(&mark);
+
+      if (!filterx_object_repr(condition_value, buf))
+        {
+          LogMessageValueType t;
+          if (!filterx_object_marshal(condition_value, buf, &t))
+            g_assert_not_reached();
+        }
+
+      msg_trace("FILTERX  COND",
```

Comment:
[optional]

I would suggest to code the result to the trace log's message for two reasons:
1. It makes it easier to find a truthy condition.
2. It becomes the same length as and `ESTEP`, `ERROR`.

```diff
-      msg_trace("FILTERX  COND",
+      msg_trace(filterx_object_truthy(condition_value) ? "FILTERX CONDT" : "FILTERX CONDF",
```

Reply:
good idea, fixed.